Binance.Net/Clients/GeneralApi/BinanceRestClientGeneralApiGiftCard.cs

```diff
@@ -0,0 +1,139 @@
+using Binance.Net.Objects.Models;
+using Binance.Net.Objects.Models.Spot.GiftCard;
+using CryptoExchange.Net.RateLimiting.Guards;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Binance.Net.Clients.GeneralApi
+{
+    /// <inheritdoc />
+    internal class BinanceRestClientGeneralApiGiftCard : IBinanceRestClientGeneralApiGiftCard
+    {
+        private static readonly RequestDefinitionCache _definitions = new();
+
+        private readonly BinanceRestClientGeneralApi _baseClient;
+
+        internal BinanceRestClientGeneralApiGiftCard(BinanceRestClientGeneralApi baseClient)
+        {
+            _baseClient = baseClient;
+        }
+
+        #region Create a single token gift card
+
+        /// <inheritdoc />
+        public async Task<WebCallResult<BinanceGiftCardResponse<BinaceGiftCardData>>> CreateSingleTokenGiftCard(string token, double amount, long? receiveWindow = null, CancellationToken ct = default)
+        {
+            var parameters = new ParameterCollection();
+            parameters.Add("token", token);
+            parameters.Add("amount", amount);
+            parameters.AddOptionalString("recvWindow", receiveWindow ?? (long)_baseClient.ClientOptions.ReceiveWindow.TotalMilliseconds);
+
+            var request = _definitions.GetOrCreate(HttpMethod.Post, "sapi/v1/giftcard/createCode", BinanceExchange.RateLimiter.SpotRestIp, 1, true);
+            return await _baseClient.SendAsync<BinanceGiftCardResponse<BinaceGiftCardData>>(request, parameters, ct).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region Create a dual token gift card
+
+        /// <inheritdoc />
+        public async Task<WebCallResult<BinanceGiftCardResponse<BinaceGiftCardData>>> CreateDualTokenGiftCard(string baseToken, string faceToken, double baseTokenAmount, long? receiveWindow = null, CancellationToken ct = default)
+        {
+            var parameters = new ParameterCollection();
+            parameters.Add("token", baseToken);
+            parameters.Add("faceToken", faceToken);
+            parameters.Add("baseTokenAmount", baseTokenAmount);
+            parameters.AddOptionalString("recvWindow", receiveWindow ?? (long)_baseClient.ClientOptions.ReceiveWindow.TotalMilliseconds);
+
+            var request = _definitions.GetOrCreate(HttpMethod.Post, "sapi/v1/giftcard/buyCode", BinanceExchange.RateLimiter.SpotRestIp, 1, true);
+            return await _baseClient.SendAsync<BinanceGiftCardResponse<BinaceGiftCardData>>(request, parameters, ct).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region Redeem a Binance Gift card
+
+        /// <inheritdoc />
+        public async Task<WebCallResult<BinanceGiftCardResponse<BinanceGiftCardRedeemData>>> RedeemGiftCard(string code, string? externalUid = null, bool useEncryption = true, long? receiveWindow = null, CancellationToken ct = default)
+        {
+            var parameters = new ParameterCollection();
+            #if NETSTANDARD2_0
```

Comment:
I think there should be an error response if using NETSTANDARD2_0 and setting `useEncryption` as true as it's not something that can be done, instead of sending it unencrypted and ignoring the parameter